File: apps/mobile/lib/domain/entities/recurring_rule.dart
enum RecurringFrequency {
  monthly,
  weekly,
  daily,
}

class RecurringRule {
  final int id;
  final int? groupId;
  final int createdBy;
  final DateTime startDate;
  final RecurringFrequency frequency;
  final String dayRule;
  final int amount;
  final int? categoryId;
  final String? merchant;
  final String? memo;
  final bool isActive;
  final DateTime createdAt;
  final DateTime updatedAt;

  RecurringRule({
    required this.id,
    this.groupId,
    required this.createdBy,
    required this.startDate,
    required this.frequency,
    required this.dayRule,
    required this.amount,
    this.categoryId,
    this.merchant,
    this.memo,
    required this.isActive,
    required this.createdAt,
    required this.updatedAt,
  });

  RecurringRule copyWith({
    int? id,
    int? groupId,
    int? createdBy,
    DateTime? startDate,
    RecurringFrequency? frequency,
    String? dayRule,
    int? amount,
    int? categoryId,
    String? merchant,
    String? memo,
    bool? isActive,
    DateTime? createdAt,
    DateTime? updatedAt,
  }) {
    return RecurringRule(
      id: id ?? this.id,
      groupId: groupId ?? this.groupId,
      createdBy: createdBy ?? this.createdBy,
      startDate: startDate ?? this.startDate,
      frequency: frequency ?? this.frequency,
      dayRule: dayRule ?? this.dayRule,
      amount: amount ?? this.amount,
      categoryId: categoryId ?? this.categoryId,
      merchant: merchant ?? this.merchant,
      memo: memo ?? this.memo,
      isActive: isActive ?? this.isActive,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
    );
  }
}

File: apps/api/src/domain/models/recurring_rule.rs
use chrono::{NaiveDate, NaiveDateTime};
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(Debug, Clone, Copy, Serialize, Deserialize, sqlx::Type)]
#[sqlx(type_name = "enum", rename_all = "UPPERCASE")]
pub enum RecurringFrequency {
    Monthly,
    Weekly,
    Daily,
}

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct RecurringRule {
    pub id: i64,
    pub group_id: Option<i64>,
    pub created_by: i64,
    pub start_date: NaiveDate,
    #[sqlx(rename = "frequency")]
    pub recurring_frequency: RecurringFrequency,
    pub day_rule: String,
    pub amount: i64,
    pub category_id: Option<i64>,
    pub merchant: Option<String>,
    pub memo: Option<String>,
    pub is_active: bool,
    #[sqlx(default)]
    pub created_at: NaiveDateTime,
    #[sqlx(default)]
    pub updated_at: NaiveDateTime,
}

